db/version_set.cc

```diff
@@ -1280,11 +1280,9 @@ void LevelIterator::Seek(const Slice& target) {
                                           ts_sz);
       if (prefix_extractor_->InDomain(target_user_key_without_ts) &&
           (!prefix_extractor_->InDomain(next_file_first_user_key_without_ts) ||
-           user_comparator_.CompareWithoutTimestamp(
```

Comment:
Thanks for the fix. I think the original code should work fine too. A prefix is a continuous group, when determine if two groups are actually the same group, it should be ok to use the user comparator's logic. 

This is checking the result to be equal or not, which is the same as what `BytewiseComparator` does and what `ReverseBytewiseComparator` does: https://github.com/facebook/rocksdb/blob/b800b5eb6a0b4934955156feab7ccc4ff9a5f3f6/util/comparator.cc#L37
https://github.com/facebook/rocksdb/blob/b800b5eb6a0b4934955156feab7ccc4ff9a5f3f6/util/comparator.cc#L160

I'm checking this since this seems to be a very hot code path, if something is wrong with it, it would be great to include a unit test to show case what bug this fixes.

Reply:
Hello, the original code performs well in the two comparators you provided. But what I want to express is that the comparator is not responsible for comparing the values of the prefix (at least there is no such requirement in its comments). If it is a user-defined comparator, it may not support comparing prefixes, resulting in errors. Actually, our custom comparator does not support comparing prefixes, so an error occurred here.

Reply:
Just want to understand the issue better, would you mind giving an example of what the custom comparator is doing and what kind of error is encountered for comparing which prefixes etc?